test/server/cards/01_SOR/units/BibFortunaJabbasMajordomo.spec.ts

```diff
@@ -0,0 +1,81 @@
+describe('Bib Fortuna', function() {
+    integration(function(contextRef) {
+        describe('Bib Fortuna\'s activated ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['bib-fortuna#jabbas-majordomo', 'jawa-scavenger'],
+                        hand: ['repair', 'confiscate', 'swoop-racer', 'surprise-strike'],
+                        base: 'capital-city',
+                        leader: 'grand-inquisitor#hunting-the-jedi',
+                    },
+                });
+            });
+
+            it('should allow the controller to play an event with a discount of 1', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.bibFortuna);
+
+                expect(context.player1).toHaveEnabledPromptButtons(['Attack', 'Play an event from your hand. It costs 1 less.']);
+                context.player1.clickPrompt('Play an event from your hand. It costs 1 less.');
+
+                expect(context.player1).toBeAbleToSelectExactly([context.repair, context.confiscate, context.surpriseStrike]);
+                context.player1.clickCard(context.repair);
+                // selects target for repair
+                context.player1.clickCard(context.capitalCity);
+
+                expect(context.bibFortuna.exhausted).toBe(true);
+                expect(context.player1.exhaustedResourceCount).toBe(0);
+
+
+                context.player2.passAction();
+
+                // cost discount from bib fortuna should be gone
+                context.player1.clickCard(context.confiscate);
+                expect(context.player1.exhaustedResourceCount).toBe(1);
+
+                context.player2.passAction();
+                context.bibFortuna.exhausted = false;
+            });
+
+            it('should not give the next unit played by the controller a discount after the controller declines to play a unit with the ability', function() {
```

Comment:
These could be combined into one test by unexhausting Bib Fortuna in between, see Rukh tests for an example of the format we tend to use (if I haven't already explained why we do this even though it's bad test practice please ping me and I can walk through it)